fix: Add column totals to feedback (#4727)

richsalz opened the pull request:
Create a new aggregate parameter that holds the totals and pass that into the HTML file as a parameter for display.
Not easy to test given #4627.

richsalz committed:
fix: Add column totals to feedback (#4727)

Create a new aggregate parameter that holds the totals and
pass that into the HTML file as a parameter for display.

rjsparks commented:
Better to test it with something in nomcom/tests.py

larseggert commented on ietf/templates/nomcom/view_feedback.html:
```diff
@@ -54,6 +54,12 @@ <h3 class="mt-5" id="declined">Declined each nominated position</h3>
                     </tr>
                 {% endfor %}
             </tbody>
+	    <tr>
+		<td></td>
+		<td></td>
+		<td><b>Totals</b></td>
+		{% for fbtot in feedback_totals %}<td>{{ fbtot }}</td>{% endfor %}
+	    </tr>
```
Should this be in a `tfoot`?

rjsparks replied:
yes, please make that change, consider `<th>` instead of the explicit `<b>`

richsalz replied:
done.

rjsparks commented on ietf/nomcom/views.py:
```diff
@@ -782,12 +782,14 @@ def nominee_staterank(nominee):
 
     sorted_nominees = sorted(nominees,key=lambda x:x.staterank)
 
+    fbtotals = []
```
This is a list, but below you treat it as a dict.
What you want here is a defaultdict.
```
>>> from collections import defaultdict
>>> totals=defaultdict(lambda:0)
>>> totals['foo']
0
>>> totals['bar']+=1
>>> totals['bar']
1
```

richsalz committed:
Use dictionary; and tfoot markup

rjsparks commented:
This didn't survive the automated change of branch target - I can work to straighten it out, or, if your local branch rebases easily, perhaps submit this again? (with apologies for the hassle).

richsalz commented:
I'll figure something out; you have enough to do.

richsalz commented:
fixed, please take a look.

rjsparks commented on ietf/nomcom/views.py:
```diff
@@ -782,12 +782,14 @@ def nominee_staterank(nominee):
         for ft in nominee_feedback_types:
             qs = nominee.feedback_set.by_type(ft.slug)
             count = qs.count()
+            fbtotals[ft.slug] = count + fbtotals.get(ft.slug, 0);
```
```suggestion
            fbtotals[ft.slug] += count
```

rjsparks commented:
I think URL mapping was about a different PR?

richsalz committed:
Use +=, per Robert

richsalz committed:
Add defaultdict import; use a filter

The fieldname is a variable, so we add a filter, `lookup` to do a
dynamic dictionary lookup.

richsalz commented on ietf/templates/nomcom/view_feedback.html:
```diff
@@ -54,6 +54,14 @@ <h3 class="mt-5" id="declined">Declined each nominated position</h3>
+		<td></td>
+		<th>Totals</h></th>
+		{% for ft in nominee_feedback_types %}
+		  <td>{{ feedback_totals|lookup:ft }}</td>
```
This line doesn't seem to work :(

richsalz replied:
Of course that fixed it.  I missed the deref in the previous `for ft in` loops :(
Still need to write a test.

dependabot[bot] committed:
ci: bump codecov/codecov-action from 3.1.1 to 3.1.2 (#5500)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from 3.1.1 to 3.1.2.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v3.1.1...v3.1.2)

---
updated-dependencies:
- dependency-name: codecov/codecov-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

richsalz committed:
Change registration call, hopefully

richsalz committed:
Use 'ft.slug', per Robert

larseggert commented on ietf/templates/nomcom/view_feedback.html:
```diff
@@ -54,6 +54,14 @@ <h3 class="mt-5" id="declined">Declined each nominated position</h3>
+	    <tfoot>
+		<td></td>
+		<td></td>
+		<th>Totals</h></th>
```
th needs scope

richsalz replied:
Do you think `col` is right?  I didn't  so I left it out

larseggert commented on ietf/templates/nomcom/view_feedback.html:
```diff
@@ -54,6 +54,14 @@ <h3 class="mt-5" id="declined">Declined each nominated position</h3>
+	    <tfoot>
+		<td></td>
+		<td></td>
+		<th>Totals</h></th>
```
```suggestion
		<th scope="row">Totals</th>
```

richsalz committed:
Update ietf/templates/nomcom/view_feedback.html

Add scope to `th`, per Lars.

Co-authored-by: Lars Eggert <lars@eggert.org>

richsalz commented:
Replaced by #5977
